### ISSUE TITLE
Frontend: Detect when user closed authorization modal (without flow being completed)

### ISSUE THREAD

**Issue by rguldener** [NangoHQ/nango#571]
Currently we do not detect when the user has closed the authorization modal without completing the authorization flow.
For many frontend UX use cases this would be useful information to have.

In brief, the reason we don't offer this is because we do not have a reliable way to detect the window close event.
More details below.

**Background**
We used to have this feature but removed it because we cannot guarantee it always works.

Here is why:
When you run `nango.auth` we open a new browser window with a builtin browser API (`window.open(...`. By default this returns us a reference to the new window.
This reference very is useful because it allows us to understand what happens in the new window we opened: We can see if it is open, read the current URL etc.

But for security reasons browsers allow the website that is being loading to cut this tie (this is called `noopener` policy). When they do this, we lose the connection to our opened window and can no longer know if it is open, has been closed etc.
Unfortunately some OAuth service providers use a `noopener` policy when they load the authorization screen to protect their users.

This means that we cannot reliably detect if the window has been closed by the user, or if a `noopener` policy has cut our access. Both look the same from the perspective of our main window.

**Potential solutions**
* Create a list of known APIs with `noopener` policy and possibly handle these in the frontend SDK
* Create a window closed detection based on timeouts
* Maybe there are other browser APIs we could leverage to know if the window is still open or was actively closed by the user?

If you have ideas feel free to comment here :)

**Comment by andyb-novo** [comment]
Maybe you can do it using eventListener in the original tab when opening the new one, to let them talk with each other?

**Comment by rguldener** [comment]
This is what we used to do @andyb-novo , the problem is that `noopener` kills that listener.
So this approach works for most APIs, but not all of them (which would be unexpected unless we handle it somehow).

Maybe we could make this a flag in the `providers.yaml` and then the Nango frontend could behave differently (disable this feature for this one flow).

**Comment by t1mmen** [comment]
Perhaps this'll spawn some ideas: https://gist.github.com/t1mmen/ef0cb5748532f2a7713da8eff54acdeb

We've been using this to avoid Safari popup blocker, and detect when the popup closes.

**Comment by rguldener** [comment]
Oh cool, thanks for sharing @t1mmen !🙏🏻

**Comment by t1mmen** [comment]
FYI, docs @  https://docs.nango.dev/sdks/frontend#triggering-oauth-flows makes it seem like this use-case is possible already using something ala..

```ts
nango.auth("<INTEGRATION-ID>", "<CONNECTION-ID>").then(...).catch((error) => {
// ... but nothing happens here on "closed popup" using either 0.17.7 or 0.21.14.
});
```

Cursory glance at the code indicates it's not yet supported, and no XHR/WebSocket activity indicates the frontend is being notified.
